app/javascript/util/listingUtil.ts

```diff
@@ -192,17 +194,36 @@ export const deriveIncomeFromAmiCharts = (
   })
 
   if (occupancyAmi) {
-    return Math.floor(occupancyAmi?.amount / 12)
+    if (min) {
+      return Math.ceil(occupancyAmi?.amount / 12)
+    }
+    return Math[min ? "ceil" : "floor"](occupancyAmi?.amount / 12)
   }
 
   return null
 }
 
-export const applyMaxIncomeToUnit =
-  (amiCharts: RailsAmiChart[]) =>
-  (unit: RailsUnitWithOccupancy): RailsUnitWithOccupancyAndMaxIncome => {
+/**
+ *
+ * For rental listings, we ignore the minimum AMI Value in ~all~ cases, including 415.
+ * For Sale listings, we try to derive the minimum AMI, if we can't then we return -1.
+ */
+const determineMinIncomeNeeded = (
+  unit: RailsUnitWithOccupancy,
+  amiCharts: RailsAmiChart[],
+  isSale: boolean
+) => {
+  return isSale
+    ? deriveIncomeFromAmiCharts(unit, unit.occupancy, amiCharts, true) || -1
+    : unit?.BMR_Rental_Minimum_Monthly_Income_Needed || -1
```

Comment:
Just wondering if there was a particular reason to choose `-1`, as opposed to `0` or just letting falsy values go through.

Reply:
For sure! I think there are a lot of correct answers here, but the reason I didn't pick a falsy value was because we use this value to calculate the absolute min/max in the header of the accordion and I didn't want to throw a falsy value into the calculation (though I _think_ it would still work, it just felt more confusing that necessary). 0 could work here, but it just seemed logically confusing since a value _could_ be zero so it would be difficult to tell if the value was zero of we just didn't have a value to start with.

Happy to reconsider this decision if you feel strongly either way

Reply:
No preference from me, I was just curious